redis/src/main/scala/com/avsystem/commons/redis/monitoring/SentinelStateObserver.scala

```diff
@@ -0,0 +1,8 @@
+package com.avsystem.commons
+package redis.monitoring
+
+import com.avsystem.commons.redis.RedisNodeClient
+
+trait SentinelStateObserver extends ConnectionStateObserver {
+  def onMasterChange(master: RedisNodeClient): Unit
```

Comment:
Why do you need to return the whole client with its internals in this callback?

I'm missing the usage in https://gitlab.avsystem.com/csp/ump/ump/-/merge_requests/5096 so really hard to tell.

Reply:
I'm 99% sure `NodeAddress` will suffice. Going to ask @pzareba22 though.